framework/src/utils/RandomData.C

```diff
@@ -96,32 +96,29 @@ RandomData::updateGenerators()
   }
 
   if (_is_nodal)
-  {
-    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
-    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
+    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
```

Comment:
There are already notes in there about this being deficient for parallel
mesh. That has been known since the start. I don't think it really matters
though, it just means you'd get different numbers in that case. I'm open to
ideas but keep in mind where we came from and why we have this design.
Without the two-level generator you can end up with patterns in your random
numbers.

There is certainly an issue with scalability in time here but not space
anymore.
On Wed, Mar 16, 2016 at 10:01 AM Derek Gaston <notifications@github.com>
wrote:

> In framework/src/utils/RandomData.C
> <https://github.com/idaholab/moose/pull/6591#discussion_r56361061>:
>
> > @@ -96,32 +96,29 @@ RandomData::updateGenerators()
> >    }
> >
> >    if (_is_nodal)
> > -  {
> > -    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
> > -    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
> > +    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
>
> Wow - I just really looked at the code for the first time... the current
> implementation is just not right. Relying on iterating over the whole mesh
> in the same way on every processor just isn't right.
>
> It shouldn't hold up this PR (this is definitely an improvement). But this
> needs to get redesigned.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6591/files/a20165253410cfe9bfc05d3c3b04e1da92707c03#r56361061>
>